src/main/java/hudson/plugins/cobertura/CoberturaPublisher.java

```diff
@@ -306,46 +308,34 @@ public void setFailingTarget(CoverageTarget failingTarget) {
      * Gets the directory where the Cobertura Report is stored for the given project.
      */
     /*package*/
-    static File getCoberturaReportDir(AbstractItem project) {
-        return new File(project.getRootDir(), "cobertura");
-    }
-
-    /**
-     * Gets the directory where the Cobertura Report is stored for the given project.
-     */
-    /*package*/
-    static File[] getCoberturaReports(AbstractBuild<?, ?> build) {
+    static File[] getCoberturaReports(Run<?, ?> build) {
         return build.getRootDir().listFiles(COBERTURA_FILENAME_FILTER);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) 
+    public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, final TaskListener listener)
             throws InterruptedException, IOException {
         Result threshold = onlyStable ? Result.SUCCESS : Result.UNSTABLE;
         if (build.getResult().isWorseThan(threshold)) {
             listener.getLogger().println("Skipping Cobertura coverage report as build was not " + threshold.toString() + " or better ...");
-            return true;
+            return;
         }
 
         listener.getLogger().println("[Cobertura] Publishing Cobertura coverage report...");
-        final FilePath[] moduleRoots = build.getModuleRoots();
-        final boolean multipleModuleRoots =
-                moduleRoots != null && moduleRoots.length > 1;
-        final FilePath moduleRoot = multipleModuleRoots ? build.getWorkspace() : build.getModuleRoot();
         final File buildCoberturaDir = build.getRootDir();
         FilePath buildTarget = new FilePath(buildCoberturaDir);
 
         FilePath[] reports = new FilePath[0];
         try {
-            reports = moduleRoot.act(new ParseReportCallable(coberturaReportFile));
+            reports = workspace.act(new ParseReportCallable(coberturaReportFile));
 
             // if the build has failed, then there's not
             // much point in reporting an error
             if (build.getResult().isWorseOrEqualTo(Result.FAILURE) && reports.length == 0) {
```

Comment:
This will likely also need the null check, however I can't test that without going in and editing the previous issue myself